src/storage/src/hummock/iterator/mod.rs

```diff
@@ -130,6 +130,15 @@ pub trait HummockIterator: Send + Sync + 'static {
     ///   before starting iteration.
     fn seek<'a>(&'a mut self, key: FullKey<&'a [u8]>) -> Self::SeekFuture<'a>;
 
+    /// Same as `seek`, except that parameter `upper_bound_key_hint` might be used to do prefetch.
+    fn seek_with_upper_bound_hint<'a>(
```

Comment:
I do not think it is necessary to add a new method to the trait `Iterator`.  Since we could pass `IteratorReadOptions` to every sub-iterator, we can set `upper_bound` and `lower_bound` and `prefetch` flag in this options.

Reply:
> I do not think it is necessary to add a new method to the trait `Iterator`. Since we could pass `IteratorReadOptions` to every sub-iterator, we can set `upper_bound` and `lower_bound` and `prefetch` flag in this options.

Shall we have a discussion? c.c. @hzxa21

Reply:
+1. I think we can pass the prefetch option when we create the iterator.